code/game/machinery/teleporter/projector.dm

```diff
@@ -27,24 +65,119 @@
 			setDir(get_dir(src, found_pad))
 			break
 
+	for(var/d in GLOB.cardinal)
+		var/turf/T = get_step(src, d)
+		for(var/obj/machinery/power/terminal/term in T)
+			if(term && term.dir == turn(d, 180))
+				terminal = term
+				break
+	if(!terminal)
+		return
+	if(!terminal.powernet)
+		terminal.connect_to_network()
+
 /obj/machinery/tele_projector/update_icon()
 	cut_overlays()
 	if(engaged)
-		update_use_power(USE_POWER_ACTIVE)
 		var/image/I = image(icon, src, "[initial(icon_state)]_active_overlay")
 		I.plane = ABOVE_LIGHTING_PLANE
 		I.layer = ABOVE_LIGHTING_LAYER_MAIN
 		add_overlay(I)
 	else
-		update_use_power(USE_POWER_IDLE)
 		if(operable())
 			var/image/I = image(icon, src, "[initial(icon_state)]_idle_overlay")
 			I.plane = ABOVE_LIGHTING_PLANE
 			I.layer = ABOVE_LIGHTING_LAYER_MAIN
 			add_overlay(I)
 
-/obj/machinery/tele_projector/attackby(obj/item/W)
-	attack_hand()
+/obj/machinery/tele_projector/attackby(obj/item/W, mob/living/user)
```

Comment:
dont you want a way to exchange SMES coils?

Reply:
maybe.
machines need a Pretty Big refactor in terms of machine parts.